### PR TITLE
session: support 'GLOBAL SCOPE' for `tidb_enable_table_partition`

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
This PR tries to support global scope for TiDB system variable `tidb_enable_table_partition`. Sometimes users may need to disable `TABLE PARTITIONS` for a TiDB cluster, at this time, setting `tidb_enable_table_partition` to `off` at global scope will be convenient.

### What is changed and how it works?
Add missing scope.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

```
tidb> show variables like '%partition%';                                                                                                                                                    +-----------------------------+-------+
| Variable_name               | Value |
+-----------------------------+-------+
| tidb_enable_table_partition | on    |
+-----------------------------+-------+
1 row in set (0.01 sec)

tidb> set global tidb_enable_table_partition=off;
Query OK, 0 rows affected (0.00 sec)

tidb> select @@global.tidb_enable_table_partition;
+--------------------------------------+
| @@global.tidb_enable_table_partition |
+--------------------------------------+
| off                                  |
+--------------------------------------+

tidb> select @@tidb_enable_table_partition;
+-------------------------------+
| @@tidb_enable_table_partition |
+-------------------------------+
| on                            |
+-------------------------------+
1 row in set (0.00 sec)

(Start a new session)
tidb> select @@tidb_enable_table_partition;
+-------------------------------+
| @@tidb_enable_table_partition |
+-------------------------------+
| off                           |
+-------------------------------+

tidb> CREATE TABLE e2 (
    ->     id INT NOT NULL,
    ->     fname VARCHAR(30),
    ->     lname VARCHAR(30),
    ->     hired DATE NOT NULL DEFAULT '1970-01-01',
    ->     separated DATE NOT NULL DEFAULT '9999-12-31',
    ->     job_code INT NOT NULL,
    ->     store_id INT NOT NULL
    -> )
    -> 
    -> PARTITION BY RANGE (store_id) (
    ->     PARTITION p0 VALUES LESS THAN (6),
    ->     PARTITION p1 VALUES LESS THAN (11),
    ->     PARTITION p2 VALUES LESS THAN (16),
    ->     PARTITION p3 VALUES LESS THAN (21)
    -> );
Query OK, 0 rows affected, 1 warning (0.02 sec)

tidb> show warnings;
+---------+------+------------------------------------------------------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                                                                  |
+---------+------+------------------------------------------------------------------------------------------------------------------------------------------+
| Warning | 8200 | Partitions are ignored because Table Partition is disabled, please set 'tidb_enable_table_partition' is if you need to need to enable it |
+---------+------+------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

tidb> explain select * from e2 where store_id > 15 ;
+---------------------+----------+-----------+-------------------------------------------------------------+
| id                  | count    | task      | operator info                                               |
+---------------------+----------+-----------+-------------------------------------------------------------+
| TableReader_7       | 3333.33  | root      | data:Selection_6                                            |
| └─Selection_6       | 3333.33  | cop[tikv] | gt(test.e2.store_id, 15)                                    |
|   └─TableScan_5     | 10000.00 | cop[tikv] | table:e2, range:[-inf,+inf], keep order:false, stats:pseudo |
+---------------------+----------+-----------+-------------------------------------------------------------+
3 rows in set (0.01 sec)
```

Code changes

 - Has exported variable/fields change

Side effects

 - NA

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release note

 - Support 'GLOBAL SCOPE' for `tidb_enable_table_partition`

### CODE REVIEW

**Comment by bb7133** [comment]
> LGTM
> 
> ```
> tidb> show warnings;                                                                                                                                                                        +---------+------+---------------------------------------------------+
> | Level   | Code | Message                                           |
> +---------+------+---------------------------------------------------+
> | Warning | 8200 | Unsupported partition type, treat as normal table |
> +---------+------+---------------------------------------------------+
> ```
> 
> the warning here is not suitable~

Addressed, PTAL

File: session/session_test.go [hunk 1]
```diff
@@ -2493,6 +2493,22 @@ func (s *testSessionSuite2) TestCommitRetryCount(c *C) {
 	c.Assert(err, NotNil)
 }
 
+func (s *testSessionSuite2) TestEnablePartition(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
```
Comment (tiancaiamao) [inline session/session_test.go]: `select @@tidb_enable_table_partition` here returns off?

Comment (bb7133) [reply session/session_test.go]: Thanks, I will set the initial value for the upgradings.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**